### DISCUSSION TITLE
Solve a time-independent problem

### DISCUSSION THREAD

**Discussion by HaukeWittich** [zwicker-group/py-pde#497] [General]
Is there a way to solve time-independend problem with py-pde. I would like to solve the Reynolds equation (https://en.wikipedia.org/wiki/Reynolds_equation) for the case that rho is constant in time. This yields to a system of linear equation for the unknown pressure which is easy to solve with e.g. the numpy tools. But I need to have access to the matrices. Are there any examples or ideas in this direction. Thank you.

**Comment by david-zwicker** [comment]
There is no special approach for time-independent problems (yet). If we encountered such problems, we reformulated them such that temporal relaxation yields a solutions and then did that. If you know of any good, general methods to solve time-independent methods, we could think about an implementation, but I'm afraid that solutions might not be unique, so methods are necessarily specialized.

**Comment by HaukeWittich** [comment]
No, sorry I do not have a general method. But perhaps, this is not needed. I guess and please correct if I am wrong, that you do a discretization of your grid (here for the scalar pressure p) like:
 
<img width="293" alt="image" src="https://github.com/zwicker-group/py-pde/assets/81968605/d8812614-deab-4a3b-98d2-6103d5c6f803">

for different derivatives. This leads to a matrice for example for the gradient in x-direction like:

<img width="503" alt="image" src="https://github.com/zwicker-group/py-pde/assets/81968605/e87eb541-2584-43bc-8069-134b6e6d73ee">

plus you have to incorporate the boundary conditions. My question was, does py-pde work like that and how can I get access to the matrices?  Because with these matrices I only have to solve these linear equation:

<img width="272" alt="image" src="https://github.com/zwicker-group/py-pde/assets/81968605/afe1ef11-8927-44e3-8a7e-d4979218f348">

**Reply by david-zwicker** [reply]
You can only express the right hand side using a matrix if it is a linear equation, which does not seem to be the case for your equation.

**Comment by HaukeWittich** [comment]
Ok, let's do it this way: 0={}*p-K. The zero means, no change in time. Now it is on the right-hand side :-)

**Reply by david-zwicker** [reply]
I don't understand what you mean here.

**Comment by HaukeWittich** [comment]
In one of the examples there is also no time deravative: Solving Poisson’s equation in 1d. I could not find how the function: solve_poisson_equation works. Any ideas?

**Reply by david-zwicker** [reply]
Yes, this is a special function that can only evaluate the Poisson equation and does not use the main machinery of `py-pde`. We added it mostly to for testing purposes.
